File: static/CWE-798/python/patched.py
# Remediation for CWE-798: Use of Hard-coded Credentials
# Fix: Load credentials from environment variables instead of hard-coding them in source.

import os
import hmac
import requests

DB_PASSWORD = os.environ["DB_PASSWORD"]
API_KEY = os.environ["API_KEY"]


def authenticate(username, password):
    return username == "admin" and hmac.compare_digest(password, DB_PASSWORD)


def call_api():
    return requests.get(
        "https://api.example.com/data",
        headers={"Authorization": "Bearer " + API_KEY},
        timeout=10,
    )

File: static/CWE-798/javascript/patched.js
// Remediation for CWE-798: Use of Hard-coded Credentials
// Fix: Load the JWT signing secret from an environment variable instead of hard-coding it in source.

const express = require('express');
const jwt = require('jsonwebtoken');

const app = express();
app.use(express.json());

const SECRET_KEY = process.env.JWT_SECRET;
if (!SECRET_KEY) {
  throw new Error('JWT_SECRET environment variable is not set');
}

app.post('/login', (req, res) => {
  const username = req.body.username;
  const token = jwt.sign({ username }, SECRET_KEY, { expiresIn: '1h' });
  res.json({ token });
});

app.get('/verify', (req, res) => {
  const decoded = jwt.verify(req.query.token, SECRET_KEY);
  res.json(decoded);
});

app.listen(3000);
